### DISCUSSION TITLE
Signals on short / angled sections of track (or path signals)

### DISCUSSION THREAD

**Discussion by pippin88** [voxeltycoon/issues#656] [General]
Currently signals can't be placed on short angled tracks.

Example:
Parallel tracks with 3 empty track spacing between.
Join the tracks - for passing etc
Not possible to place a signal on joining track. Intersection is blocked for both tracks.
![Signal Limitation](https://user-images.githubusercontent.com/6128210/116657944-98664e80-a9d2-11eb-8493-700e7a34504e.jpg)

Having 4 block spacing between tracks does allow a signal to be placed.
But this takes up a **lot** of room
![Signal Limitation 1](https://user-images.githubusercontent.com/6128210/116658160-e67b5200-a9d2-11eb-864f-ebe43733b17e.jpg)

If path signals are not going to be implemented, it would be great to be able to put signals on angles tracks / shorter bits

**Comment by Road-Train** [comment]
Your problem could be fixed by replacing your bottom left signal with a pre-signal. Still wouldn't quite work as intended, but it is a solution which works a lot better. I'm up for allowing (pre) signals to be placed on the edges of two diagonals though.

**Reply by pippin88** [reply]
That doesn't fix the problem.

The problem is that Voxel Tycoon uses simple block signalling, and doesn't allow signals on curves that join a straight.

Changing any of the signals in the first picture to a pre-signal doesn't change the fact it's still in one block. Both tracks will be blocked by a train on one track.

If signals could be placed on diagonals as they come off straights it would be much less of an issue.

**Comment by hkoskenv** [comment]
I agree and hope also free signal placement. It is for aesthetic reasons to get more realistic looking intersections and railyards. Now I have to leave huge gaps between rails to leave room for signals which make things look ugly. As far as I know there is no technical or game balancing reasons to need large free area around the signal.
